### PR TITLE
Update minimum required python version to 3.5.2

### PR DESCRIPTION
**Pull request by take-cheeze**
Rel: https://github.com/chainer/chainer/pull/8405

### CODE REVIEW

File: docs/source/upgrade.rst [hunk 1]
```diff
@@ -31,7 +31,7 @@ Dropping Support of Python 3.4
 ------------------------------
 
 In Chainer v6, Python 3.4 is no longer supported as it reaches its end-of-life (EOL) in March 2019.
-Python 3.5.1 is the minimum Python 3 version supported by Chainer v6.
+Python 3.5.2 is the minimum Python 3 version supported by Chainer v6.
```
Comment (kmaehashi) [inline docs/source/upgrade.rst]: Let's avoid changing requirements for v6 (1) It is a stable version which is already released (2) onnx-chainer is not backported to v6

Comment (take-cheeze) [reply docs/source/upgrade.rst]: Thank you for pointing out! Should be fixed with: https://github.com/chainer/chainer/pull/8410/commits/38a9f186b2c4f79f7987cb95cdb5c645c2615606

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: docs/source/chainermn/installation/guide.rst [hunk 2]
```diff
@@ -123,7 +123,7 @@ We tested ChainerMN on all the following environments.
   * Ubuntu 14.04 LTS 64bit
   * Ubuntu 16.04 LTS 64bit
 
-* Python 2.7.13, 3.5.1, 3.6.1
+* Python 2.7.13, 3.5.2, 3.6.1
```
Comment (kmaehashi) [inline docs/source/chainermn/installation/guide.rst]: Confirmed with ChainerMN team.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**